app/src/main/kotlin/com/adesso/movee/base/ResProvider.kt

```diff
@@ -0,0 +1,36 @@
+package com.adesso.movee.base
+
+import android.app.Application
+import androidx.annotation.ArrayRes
+import androidx.annotation.ColorRes
+import androidx.annotation.DimenRes
+import androidx.annotation.DrawableRes
+import androidx.annotation.IntegerRes
+import androidx.annotation.StringRes
+import androidx.core.content.ContextCompat
+import com.adesso.movee.application.MainApplication
+import javax.inject.Inject
+
+class ResProvider @Inject constructor(val application: Application) {
```

Comment:
Resource erişimi için application context kullanmak doğru değildir. Hem "theme" bilgisine sahip değil, hem de "configuration change" durumlarında yanlış veri verecektir. Burada "activity context" kullanılmalı ve bu sınıf "activity lifecycle scope" içerisinde yaşamalıdır. Bu yüzden viewModel içinde de kullanılamaz.

Reply:
ResProvider class'ını silip yerine ResourceProvider class'ını inject ettim. ResourceProvider içerisinde şimdilik string çekme işlemleri yapıyorum. Activity context ile stringler çekilecek şekilde güncelledim, teşekkürler.